### DISCUSSION TITLE
is there skywalking metrics api interface,we can get metrics from skywalking metrics api interface

### DISCUSSION THREAD

**Discussion by liyuntian** [apache/skywalking#10505] [Q&A]
such as,I can get Service Avg Response Time from http://xxxx:8080/graphql/Service-Avg-Response-Time/xxxxxxx

**Comment by wu-sheng** [comment]
I don't know what this means. GraphQL is defined and open for all users.

**Comment by Superskyyy** [comment]
GraphQL and gRPC exporters are there for you to either query or mass export

**Reply by liyuntian** [reply]
thank you
We want to obtain the data in metrics format according to the skywalking interface. Currently, the results obtained through the graphql interface are in json format, so we shold convert graphql interface result to metrics format result by ourselves ?

**Reply by Superskyyy** [reply]
> thank you
> 
> We want to obtain the data in metrics format according to the skywalking interface. Currently, the results obtained through the graphql interface are in json format, so we shold convert graphql interface result to metrics format result by ourselves ?

if you need out of box protobuf format, the grpc exporter can be used. Since graphql interface is for adhoc queries and frontend usage it obviously should serve in json right? Indeed you can convert, but if you need large scale analytics you should go with exporter.

**Reply by wu-sheng** [reply]
What @liyuntian asking doesn't exist in any kind of open-source system. I never saw this kind of query, and I can't see what is `metrics format result`. There isn't such a definition.

**Reply by liyuntian** [reply]
https://github.com/rebelsre/skywalking_exporter  this is what I want
